src/jsonserializer/jsonserializer.qbs

```diff
@@ -1,12 +1,30 @@
 import qbs
 import qbs.TextFile
 
-DynamicLibrary {
+Product {
     id: root
+
+    property bool isAndroid: qbs.targetOS.contains("android")
+    property bool isMacOS: qbs.targetOS.contains("macos")
+    property bool isWindows: qbs.targetOS.contains("windows")
+
     targetName: "QtJsonSerializer"
 
     Depends { name: "Qt.core" }
     Depends { name: "cpp" }
+
+    type: {
+        if (isAndroid) {
+            return "staticlibrary"
+        }
+        else if (isMacOS) {
+            return "staticlibrary"
+        }
+        else if (isWindows) {
+            return "dynamiclibrary"
+        }
```

Comment:
These conditions are non-exhaustive. There is a default case missing.
From the history I would assume:
```qbs
type: (isAndroid || isMacOs ? 'staticlibrary' : 'dynamiclibrary')
```

Reply:
we might also use one property for this:
```qbs
property bool isStaticLibrary: isAndroid || isMacOs
type: (isStaticLibrary ? 'staticlibrary' : 'dynamiclibrary')
```